lighthouse-core/gather/driver.js

```diff
@@ -717,10 +717,20 @@ class Driver {
    * @return {string}
    */
   getRequestContent(requestId) {
-    return this.sendCommand('Network.getResponseBody', {
-      requestId,
-    // Ignoring result.base64Encoded, which indicates if body is already encoded
-    }).then(result => result.body);
+    return new Promise((resolve, reject) => {
+      // If this takes more than 2s, reject the Promise.
+      const err = new Error('Fetching resource content has exceeded the allotted time of 2s');
```

Comment:
should we try to give this thing an error code and be consistent about timeout stuff moving forward?